tests/python/test_with_sklearn.py

```diff
@@ -34,7 +34,8 @@ def test_binary_classification():
     kf = KFold(n_splits=2, shuffle=True, random_state=rng)
     for cls in (xgb.XGBClassifier, xgb.XGBRFClassifier):
         for train_index, test_index in kf.split(X, y):
-            xgb_model = cls(random_state=42).fit(X[train_index], y[train_index])
+            clf = cls(random_state=42, eval_metric=['auc', 'logloss'])
+            xgb_model = clf.fit(X[train_index], y[train_index])
```

Comment:
Actually there's a `eval_metric` in `fit`.

Reply:
?? The example in #5340 was using `eval_metric` in the constructor.

Reply:
Okay, looks like
```python
model.fit(X, y, eval_metric=['auc', 'logloss'])
```
also triggers the same bug.